src/main/groovy/com/ullink/ProjectFileParser.groovy

```diff
@@ -44,21 +44,42 @@ class ProjectFileParser {
         FilenameUtils.removeExtension(file) + newExtension
     }
 
+    File getOutputPath() {
+        if (msbuild.destinationDir) {
+            project.file(msbuild.destinationDir)
+        }
+        if (properties.TargetDir) {
+            new File (properties.TargetDir)
+        }
+    }
+
     File getDotnetAssemblyFile() {
-        project.file(properties.TargetPath)

```

Comment:
yup, the following throws exception as the first item in msbuild.projects is the solution
msbuild.projects.each { _, project -> project.dotnetArtifacts.files.each { println it.path } }